### DISCUSSION TITLE
Detect if realtime event was triggered from local user or remote?

### DISCUSSION THREAD

**Discussion by aonghas** [pocketbase/pocketbase#4636] [Q&A]
When using snapshot listeners with Firestore there is a way to ignore changes that are made by the local system (as opposed to someone else making changes to the same document). Is it possible to detect something similar with Pocketbase realtime? Basically I want to ignore realtime events that are triggered by my own PATCH updates, but not updates by others on the same record.

**Comment by ganigeorgiev** [comment]
There is no tracking out of the box for who submitted a record change if that is what you are asking.

The description is too vague and lack details to understand the broader picture, but I guess one option if you want to track the users who submitted a record change is to add a `modifiedBy` users `relation` field to your collection and sync/update it programmatically with the before request hooks. Then you can use the `filter` option in your subscription to exclude the record that you've modified with something like `modifiedBy != "yourUserId"`. Here is a minimal example using the JS `pb_hooks`:

```js
onRecordBeforeCreateRequest((e) => {
    const info = $apis.requestInfo(e.httpContext);

    if (info.admin) {
        return // admin can submit anything
    }

    // always overwrite the modifiedBy value with the current authenticated user id
    if (info.authRecord) {
        e.record.set("modifiedBy", info.authRecord.id)
    }
}, "yourCollection")

onRecordBeforeUpdateRequest((e) => {
    const info = $apis.requestInfo(e.httpContext);

    if (info.admin) {
        return // admin can submit anything
    }

    // always overwrite the modifiedBy value with the current authenticated user id
    if (info.authRecord) {
        e.record.set("modifiedBy", info.authRecord.id)
    }
}, "yourCollection")

// (you can do the same for the delete too if you need to but an extra manual update will be needed)
```

Then on the client-side, you can subscribe like:
```js
await pb.collection("yourCollection").subscribe("*", (e) => {
    ...
}, {
    // (replace with the id you want to exclude)
    filter: pb.filter("modifiedBy != {:id}", {id: pb.authStore.model.id})
})
```
